scripts/build/builders/nxp.py

```diff
@@ -129,16 +143,19 @@ def __init__(self,
                  disable_ble: bool = False,
                  enable_thread: bool = False,
                  enable_wifi: bool = False,
-                 disable_ipv4: bool = False,
                  enable_shell: bool = False,
-                 enable_ota: bool = False):
+                 enable_ota: bool = False,
+                 enable_factory_data_build: bool = False,
+                 disable_pairing_autostart: bool = False,
+                 board_variant: str = None):
```

Comment:
this is not str given that its default is None. It should be at least `Optional[str]`. also see my previous notes about this being an enum.

Could you also please make patches incremental instead of force-push? Force-push seems to cause loss of history/placement of comments.